zanata-war/src/main/java/org/zanata/security/ZanataIdentity.java

```diff
@@ -141,12 +144,12 @@ public boolean hasPermission(String name, String action, Object... arg) {
         if (result) {
             if (log.isDebugEnabled()) {
                 log.debug("ALLOWED hasPermission({}, {}, {}) for user {}",
-                        name, action, arg, getAccountUsername());
+                        name, action, Lists.newArrayList(arg), getAccountUsername());
             }
         } else {
             if (log.isDebugEnabled()) {
```

Comment:
Does this check even do anything? I thought that `log.debug(...)` would implicitly perform this check anyway. Is it a performance optimization to prevent the creation of the new array list for `arg`?